### ISSUE TITLE
Bug: Natron doesn't start on WSL2/Win10 with VcXsrv

### ISSUE THREAD

**Issue by garyo** [NatronGitHub/Natron#682]
### 

- [X] I'm using the latest version of Natron (not required but recommended)
- [X] I've restarted Natron and the issue persists
- [X] I've run Natron via the [command line](https://natron.readthedocs.io/en/rb-2.4/devel/natronexecution.html) and the issue persists
- [X] I've followed the [contributing guidelines](https://github.com/NatronGitHub/Natron/blob/RB-2.4/CONTRIBUTING.md) to the best of my understanding
- [X] My issue is not on the [issue tracker](https://github.com/NatronGitHub/Natron/issues?q=is%3Aissue+type%3Abug) and is not a duplicate of a forum thread
- [X] This bug is reproducible
- [X] This issue is not a feature request or a pull request

### Natron version

Natron 2.4.0

### Operating system

WSL2 running Ubuntu 20.04 on Windows 10

### System specs

```markdown
RAM: 64GB
CPU: AMD Threadripper
GPU: 2x NVIDIA
```

### Did you install Natron using the official installer?

- [X] Yes, I used the official installer
- [ ] No, I installed from a binary archive
- [ ] No, I compiled Natron from sources
- [ ] No, I installed Natron via another method

### What were you trying to do?

```markdown
Start Natron in a WSL2 container, displaying on Windows using VcXsrv.
```

### What did you expect to happen? What happened instead?

```markdown
Expected it to run OK, at least with LIBGL_ALWAYS_SOFTWARE=1. But it doesn't start; gives these errors on the cmdline:

% LIBGL_ALWAYS_INDIRECT=1 LIBGL_ALWAYS_SOFTWARE=1 /tmp/Natron-2.4.0-Linux-64-no-installer/Natron
Natron Version 2.4
(C) 2018-2021 The Natron developers
(C) 2013-2018 INRIA and Alexandre Gauthier-Foichat
>>>Use the --help or -h option to print usage.<<<
Error while loading OpenGL: GLX: Created context is not doing direct rendering
OpenGL rendering is disabled.
GLX: Created context is not doing direct rendering
GLX: Created context is not doing direct rendering
Info: init.py script not loaded (this is not an error)
Info: initGui.py script not loaded (this is not an error)
GLX: Created context is not doing direct rendering
```
and then a dialog box where clicking OK causes Natron to exit:
```


### Step-by-step reproduction instructions

```markdown
* On Windows 10, enable and set up WSL2
* Install Ubuntu 20.04 in WSL2
* Install & start vcXsrv, configure it with (or without) native OpenGL and turn off X11 security
* Install Natron in Ubuntu
* Start Natron
* Observe errors and exit
```

### Additional details

```markdown
It does run OK in the new Windows 11 version which has a built-in X11 server, but not in Windows 10 using VcXsrv. Happy to provide more info; just ask.
```

**Comment by devernay** [comment]
Hi Gary, I didn't know you were a Natron user, and welcome!

- Is native openGL enabled or disabled in VcXsrv?
- Does "glxgears" work fine? Of course, set the same environment variables to the same values when running it.
- Can you please please attach the output of "glxinfo" (from `mesa-utils` package) and "glewinfo" (from `glew-utils`) to this ticket? Both on Win10+VcXsrv and Win11 would be nice. Same environment variables too.
- This comment mentions that one should use LIBGL_ALWAYS_INDIRECT=0 AND diable native OpenGL, did you try that as well?  https://github.com/microsoft/WSL/issues/2855#issuecomment-613935650

If none of these fixes the issue, I'm not sure we can do something about it, though.

**Comment by garyo** [comment]
Hi! Yes, I use it quite a bit these days.

* VcXsrv has native openGL enabled
* glxgears seems to work OK, both with & without LIBGL_ALWAYS_INDIRECT: 337 fps w/ indirect, 228 fps without indirect. However it does give these errors on startup:
```
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
```
I tried disabling native OpenGL, it didn't seem to make any difference to Natron. (I'll try it again though.)

On the Win10 machine with vcXrsv, where it doesn't work:
* Here's glxinfo:  [glxinfo.txt](https://github.com/NatronGitHub/Natron/files/7368933/glxinfo.txt)
* and here's [glewinfo.txt](https://github.com/NatronGitHub/Natron/files/7368923/glewinfo.txt)

I'll add the files from Win11 in a minute.

```

**Comment by garyo** [comment]
Here are the same logs from the Win11 system where Natron works OK. I note that on that system, glxgears doesn't print those errors. 

[glewinfo.txt](https://github.com/NatronGitHub/Natron/files/7369171/glewinfo.txt)
[glxinfo.txt](https://github.com/NatronGitHub/Natron/files/7369172/glxinfo.txt)

**Comment by devernay** [comment]
Thanks!
Does this impair your productivity, and do you have a need for running it under WSL2 with Win10? Or does the fallback solution (using Win11) fix the issue for you?
If that's OK with you, we could mark the issue as wontfix, and state this is an unsupported configuration. It may work again when we switch to Qt5/6 (which will not happen soon).
Anyway, thanks for testing this! I didn't even know WSL2 now had X11 graphics under Win11, this opens a new world for Windows users!

**Comment by garyo** [comment]
My main system is still Win10 (uses older Threadripper CPU & can't upgrade to Win11 for various reasons) so yes it would be very nice if I could do Linux OFX development there... but if it needs the newer OpenGL features maybe there's nothing that can be done? If there's anything I can try, please let me know.

**Comment by rodlie** [comment]
https://github.com/microsoft/WSL/discussions/6154 seems to be related.

**Comment by garyo** [comment]
@rodlie absolutely. It creates the WSLg distro that runs in the background, running Wayland, X11, as well as sound etc. But requires Win11 at this point.

Actually I was able to get Win11 on my main dev machine today! So I don't need this as much anymore. Natron works OK in that situation.
